src/se/vidstige/jadb/AdbServerLauncher.java

```diff
@@ -18,17 +18,23 @@ public AdbServerLauncher(Runtime runtime) {
 
     private String findAdbExecutable() {
         String android_home = System.getenv("ANDROID_HOME");
-        if (android_home == null || android_home.equals("")) {
-            return "adb";
-        }
-
-        return android_home + "/platform-tools/adb";
+        return android_home != null && !android_home.equals("")?android_home + "/platform-tools/adb":"adb";
     }
 
     public void launch() throws IOException, InterruptedException {
-        Process p = runtime.exec(new String[]{findAdbExecutable(), "start-server"});
+        this.action("start-server");
+    }
+
+    public void kill() throws IOException, InterruptedException {
+        this.action("kill-server");
+    }
+
+    private void action(String command) throws IOException, InterruptedException {
+        Process p = this.runtime.exec(new String[]{this.findAdbExecutable(), command});
         p.waitFor();
         int exitValue = p.exitValue();
-        if (exitValue != 0) throw new IOException("adb exited with exit code: " + exitValue);
+        if(exitValue != 0) {
+            throw new IOException("adb commands <" + command + "> exited with exit code: " + exitValue);
```

Comment:
for clearer error message lets just do `"'adb + command + "' failed with exit code" + exitValue`